src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/StoredProcedures/GetAvailableExportJobs/getAvailableExportJobs.js

```diff
@@ -0,0 +1,102 @@
+* @param {string} jobHeartbeatTimeoutThresholdInSecondsInString - The number of seconds allowed before the job is considered to be stale in string.
+*/
+
+function getAvailableExportJobs(maximumNumberOfConcurrentJobsAllowedInString, jobHeartbeatTimeoutThresholdInSecondsInString) {
+    const collection = getContext().getCollection();
+    const collectionLink = collection.getSelfLink();
+    const response = getContext().getResponse();
+
+    const errorMessages = {
+        InvalidMaximumNumberOfConcurrentJobsAllowedInString: `${ErrorCodes.BadRequest}: The specified maximumNumberOfConcurrentJobsAllowedInString with value '${maximumNumberOfConcurrentJobsAllowedInString}' is invalid.`,
+        InvalidJobHeartbeatTimeoutThresholdInSecondsInString: `${ErrorCodes.BadRequest}: The specified jobHeartbeatTimeoutThresholdInSecondsInString with value '${jobHeartbeatTimeoutThresholdInSecondsInString}' is invalid.`,
+        RequestEntityTooLarge: `${ErrorCodes.RequestEntityTooLarge}: The request could not be completed.`
+    };
+
+    // Validate input
+    if (!maximumNumberOfConcurrentJobsAllowedInString) {
+        throw new Error(errorMessages.InvalidMaximumNumberOfConcurrentJobsAllowedInString);
+    }
+
+    let maximumNumberOfConcurrentJobsAllowed = parseInt(maximumNumberOfConcurrentJobsAllowedInString);
+
+    if (maximumNumberOfConcurrentJobsAllowed <= 0) {
+        throw new Error(errorMessages.InvalidMaximumNumberOfConcurrentJobsAllowed);
+    }
+
+    if (!jobHeartbeatTimeoutThresholdInSecondsInString) {
+        throw new Error(errorMessages.InvalidJobHeartbeatTimeoutThresholdInSecondsInString);
+    }
+
+    let jobHeartbeatTimeoutThresholdInSeconds = parseInt(jobHeartbeatTimeoutThresholdInSecondsInString);
+
+    if (jobHeartbeatTimeoutThresholdInSeconds <= 0) {
+        throw new Error(errorMessages.InvalidJobHeartbeatTimeoutThresholdInSecondsInString);
+    }
+
+    // Calculate the expiration time in seconds where the job is considered to be stale.
+    let expirationTime = new Date().setMilliseconds(0) / 1000 - jobHeartbeatTimeoutThresholdInSeconds;
+
+    tryQueryRunningJobs();
+
+    function tryQueryRunningJobs() {
+        // Find list of active running jobs.
+        let query = {
+            query: `SELECT VALUE COUNT(1) FROM ROOT r WHERE r.jobRecord.status = 'Running' AND r._ts > ${expirationTime}`
+        };
+
+        let isQueryAccepted = collection.queryDocuments(
+            collectionLink,
+            query,
+            {},
+            function (err, resources) {
+                if (err) {
+                    throw err;
+                }
+
+                let numberOfRunningJobs = resources[0];
+                
+                tryQueryAvailableJobs(numberOfRunningJobs);
+            });
+
+        if (!isQueryAccepted) {
+            // We ran out of time.
+            throw new Error(errorMessage.requestEntityTooLarge);
```

Comment:
provide an error code number so you don't have to rely on strings.

Reply:
The requestEntityTooLarge is defined as `${ErrorCodes.RequestEntityTooLarge}: The request could not be completed.`, which has the error code in the message itself. Since the error object only takes string, that's what we are throwing so that the client can parse the error code out of the message and react accordingly (which I realized is missing so I will add that). Is that what you are referring to?

Reply:
You can provide an error code: `throw new Error(404, "Document not found");` then you can get the 404 from the x-ms-substatus header. An [example](https://github.com/Azure/azure-cosmos-dotnet-v3/blob/dba4b60bddafe6f6c8ad579c5d8f82c28d4e1969/Microsoft.Azure.Cosmos/src/ChangeFeedProcessor/DocDBErrors/SubStatusHelpers.cs).

Reply:
(obviously you would not use 404 here :))

Reply:
Oh i didn't know that existed. That's better than what we have right now. I will update the code.

Reply:
Updated the stored procedure to throw error with status code.